src/com/android/server/connectivity/NetworkMonitor.java

```diff
@@ -1638,9 +1639,17 @@ private void handlePrivateDnsEvaluationFailure() {
             }
         }
 
+        private boolean usingGrapheneConnectivityCheck() {
+            return (mContext.getResources().getString(R.string.config_captive_portal_https_url) ==
```

Comment:
Better to use `equals` here instead of `==` since it could have been copied at some point even if it isn't in practice. Java `==` isn't value comparison for strings or arrays due to a mistake standardizing it (and for arrays, equals is shallow comparison too, so there's Arrays.equals() instead).

Reply:
Also, this isn't the right way to fetch the current value.

Reply:
May I please know what is the right way to fetch the current value?

Reply:
I don't know but what you're fetching is a hard-wired string. You need to fetch the value of the preference. You should do it how it's done for fetching it in this NetworkStack module for using the connectivity check URL.

Reply:
Thanks! I think I got it, well assuming `getCaptivePortalServerHttpsUrl()` is the correct way.

Reply:
Can you add logging there and test that it works as intended when toggled? It performs this check every time a network is first connected when Private DNS is set to automatic.

Reply:
Seems to work based on my testing and logging with NextDNS.